### PR TITLE
Renamed TranslationFilter and DocumentFilter to Get[Translation/Document]StatusesOptions

### PR DESCRIPTION
**Pull request by AhmedLeithy**
TranslationFilter --> GetTranslationStatusesOptions
DocumentFilter --> GetDocumentStatusesOptions
as per #23088 

Should we also change TranslationFilterOrder and TranslationFilterProperty? (DocumentFilter.. as well)

TranslationFilterOrder --> GetTranslationStatusesOptionsOrder seems a bit long and confusing in my opinion, but leaving them as is might also be a bit confusing. What do you think?

### CODE REVIEW

File: sdk/translation/Azure.AI.Translation.Document/src/GetDocumentStatusesOptions.cs [hunk 1]
```diff
@@ -11,12 +11,12 @@ namespace Azure.AI.Translation.Document
     /// A class which defines filtering and ordering options
     /// for listing all document statuses for a certain translation operation.
     /// </summary>
-    public partial class DocumentFilter
+    public partial class GetDocumentStatusesOptions
```
Comment (maririos) [inline sdk/translation/Azure.AI.Translation.Document/src/GetDocumentStatusesOptions.cs:14]: with this move, I wonder if we could undo something I did 😆 .
The generated client had a class called `DocumentFilter` that was used for other purposes I didn't like, so I made it internal. https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/translation/Azure.AI.Translation.Document/src/DocumentFilterInternal.cs
Now that we are not using the public `DocumentFilter` class, we can undo that and let the generated `DocumentFilter` be internal and that's it.

Let me know if that is not clear and we can sync

**Review by maririos — APPROVED** [review]

**Comment by maririos** [comment]
>Should we also change TranslationFilterOrder and TranslationFilterProperty? (DocumentFilter.. as well)
TranslationFilterOrder --> GetTranslationStatusesOptionsOrder seems a bit long and confusing in my opinion, but leaving them as is might also be a bit confusing. What do you think?

I agree with your decision. Let's leave those types as is

File: sdk/translation/Azure.AI.Translation.Document/src/GetDocumentStatusesOptions.cs (hunk 1 above)
Comment (AhmedLeithy) [reply sdk/translation/Azure.AI.Translation.Document/src/GetDocumentStatusesOptions.cs:14]: Sure thing. So this involves allowing it to generate DocumentFilterInternal with the name DocumentFilter the same way it used to before we defined the customized model. I just need to make sure that the generated class is internal.

Got it!

**Comment by AhmedLeithy** [comment]
Without the customized model, the generated filter is already internal. So I removed `DocumentStatusInternal.cs` all together. The only difference this cause is that the model, is generated in the namespace ...Document.Models, but I don't think that would be an issue.

File: sdk/translation/Azure.AI.Translation.Document/src/GetDocumentStatusesOptions.cs (hunk 1 above)
Comment (maririos) [reply sdk/translation/Azure.AI.Translation.Document/src/GetDocumentStatusesOptions.cs:14]: thank you!!

**Review by maririos — APPROVED** [review]

**Review by jsquire — CHANGES_REQUESTED** [review]
Please rebase on main before merging.   There are a lot of unrelated changes riding along that we should remove.

**Comment by AhmedLeithy** [comment]
> Please rebase on main before merging. There are a lot of unrelated changes riding along that we should remove.

Done! Apologies.

**Review by jsquire — APPROVED** [review]

**Pull request merged**